### ISSUE TITLE
Bug: [no-throw-literal] false positive when using interface

### ISSUE THREAD

**Issue by mip29** [typescript-eslint/typescript-eslint#8024]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.1.6&fileType=.ts&code=MYGwhgzhAECiBO8D28CSBbADiaBLLIApuoQHYAuMCy80A3gFDTQlRgDmh0AvNAESFEKFoTac%2BAbibRSYEj36CaMuYT7QGAXwYNQkKkPiwAHuTIRcSUtEKmyAEwPLGWnQDMArqWDlL18gAWyADuALKEAIKk1CgmZqQWVgAUAJT00oEhMoTBcIZx5n6pUtoMnt6%2BVtCZSGGR0YYY2KnpzDW5pDl5NE0gxVpAA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQHYHsBaAFwAtp8B3Q%2BAS2MWgEN50p8AzDyAGnG0gABYgE8ADigDG0WmOKEUdXMQD0BEuSo16jFm0iMK0SPwC%2BIU0A&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
class ErrorImpl implements Error {
  message = "error message";
  name = "error name" 
}

class ErrorExtension extends Error {

}

function throwMeAnErrorExtension() {
  throw new ErrorExtension();
}

function throwMeAnErrorImpl() {
  throw new ErrorImpl();
}
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "no-throw-literal": "off",
    "@typescript-eslint/no-throw-literal": "error"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

No errors. Implementing the Error interface is a valid Error.

### Actual Result

The lint error "Expected an error object to be thrown." is thrown at line 15.

**Comment by bradzacher** [comment]
The rule enforces that a type that subclasses `Error` to be thrown.
A class implementing the `Error` interface is not a subclass of `Error`.
Implementing `Error` is no better than having an object literal.
